Automattic/wp-calypso
# Don't show upsell if a term variant of the upsold product is already owned

### Changes proposed in this Pull Request

Let's say you own a product of a specific term (e.g. Backup Daily, monthly). When purchasing another product that upsells it (e.g. Scan), with a different term selected, the upsell is still shown. In this PR, we make sure to not show the upsell in that case.

Note: upselling a product with a superior term is maybe something we'd like to do in the future, but the UI doesn't handle this case well for now.

### Testing instructions


- Select a self-hosted Jetpack site that has no product or plan
- Visit the _Plans_ section with the offer flow enabled
- Try to purchase Scan, with any term
- Verify that you see the upsell for Backup Daily
- Select a site that has Backup Daily, with a monthly term, and go back to the _Plans_ page
- Try to purchase Scan, with a yearly term
- Verify that no upsell is shown
- Do the same with Backup Daily on a yearly term, and Scan on a monthly term
